sklearn/linear_model/least_angle.py

```diff
@@ -1078,7 +1071,13 @@ def fit(self, X, y):
         # init cross-validation generator
         cv = check_cv(self.cv, X, y, classifier=False)
 
-        Gram = 'auto' if self.precompute else None
+        # As we use cross-validation, the Gram matrix is not precomputed here
+        Gram = self.precompute
+        if hasattr(Gram, '__array__'):
+            warnings.warn("Parameter 'precompute' cannot be an array in "
```

Comment:
I was thinking about avoiding an error when a user changes `Lars(precompute=G).fit(X, y)` into `LarsCV(precompute=G).fit(X, y)`. But it is kind of nursing.